### PR TITLE
Mediamtx Dockefile build improvements

### PR DESCRIPTION
**Pull request by pwilczynskiclearcode**

### CODE REVIEW

File: docker/Dockerfile.mediamtx [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+      ca-certificates \
+      curl \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN curl -L https://github.com/bluenviron/mediamtx/releases/download/${MEDIAMTX_VERSION}/mediamtx_${MEDIAMTX_VERSION}_linux_amd64.tar.gz -o /mediamtx.tar.gz \
+    && tar xzvf /mediamtx.tar.gz \
+    && rm /mediamtx.tar.gz
 
 COPY docker/mediamtx.yml /mediamtx.yml
```
Comment (hjpotter92) [inline docker/Dockerfile.mediamtx]: can set the context of docker build to be inside `docker/` if there are no other files from the repo that are needed.

**Review by hjpotter92 — APPROVED** [review]
lgtm. small comment, but not worth stopping the pr to go through currently.

File: docker/Dockerfile.mediamtx [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+    && apt-get install -y \
+        ca-certificates \
+        curl \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
-COPY docker/mediamtx.yml /mediamtx.yml
+RUN curl -L https://github.com/bluenviron/mediamtx/releases/download/${MEDIAMTX_VERSION}/mediamtx_${MEDIAMTX_VERSION}_linux_amd64.tar.gz -o /mediamtx.tar.gz \
+    && tar xzvf /mediamtx.tar.gz \
+    && rm /mediamtx.tar.gz
```
Comment (pwilczynskiclearcode) [inline docker/Dockerfile.mediamtx:15]: @mjh1  `mediamtx.tar.gz` contains `mediamtx.yml` file and it looks it differs from the one in [our repo](https://github.com/livepeer/go-livepeer/blob/master/docker/mediamtx.yml) with these lines:
```diff
< logLevel: info
---
> logLevel: debug

<   runOnReady:
---
>   runOnReady: curl http://localhost:5936/live/video-to-video/$MTX_PATH/start -F callback=$HOSTNAME -F source_id=$MTX_SOURCE_ID -F source_type=$MTX_SOURCE_TYPE -F query=$MTX_QUERY
```

Could we use [environment variables](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#configuration) for overriding the defaults instead? Like:
```
ENV MTX_LOGLEVEL=debug
ENV MTX_PATHDEFAULTS_RUNONREADY='curl http://localhost:5936/live/video-to-video/$MTX_PATH/start -F callback=$HOSTNAME -F source_id=$MTX_SOURCE_ID -F source_type=$MTX_SOURCE_TYPE -F query=$MTX_QUERY'
```

and use the default `mediamtx.yml` bundled with binary?

Comment (pwilczynskiclearcode) [reply docker/Dockerfile.mediamtx:15]: https://github.com/livepeer/livepeer-infra/pull/2478/files#r1850383090

Comment (mjh1) [reply docker/Dockerfile.mediamtx:15]: ohh nice spot 👍

Comment (j0sh) [reply docker/Dockerfile.mediamtx:15]: yes, nice one!

**Pull request merged**